library/src/main/java/org/xmtp/android/library/Client.kt

```diff
@@ -338,6 +391,7 @@ class Client() {
 
                 val encryptionKey = options.dbEncryptionKey
                     ?: throw XMTPException("No encryption key passed for the database. Please store and provide a secure encryption key.")
+                val nonce = if (privateKeyBundleV1 != null) 0UL else Random.nextULong()
```

Comment:
You can still use nonce 0 for clients without a privateKeyBundle. You just cannot use nonce > 0 for clients created with one.

In either case, I'd be very careful with generating a random nonce as part of building the client. This should only happen if there is no inbox_id associated with the wallet. Otherwise you could create multiple inboxes for the same wallet address.